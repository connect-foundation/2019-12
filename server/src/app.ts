import * as express from 'express';

const app = express();

export default app;
