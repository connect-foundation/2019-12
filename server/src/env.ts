import { config } from 'dotenv';
import { resolve } from 'path';

const NODE_ENV = process.env.NODE_ENV;
const envFileName = NODE_ENV ? `.env.${NODE_ENV}` : '.env';
const path = resolve(__dirname, `../${envFileName}`);

config({ path });
