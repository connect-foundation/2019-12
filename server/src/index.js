const express = require('express');
const app = express();
const port = 3000;

const { makeUUID } = require('./bin/uuid.js');

app.get('/', (req,res) => {
	const getUUID = makeUUID();
	res.send(getUUID());
});

app.listen(port);

module.exports = app;