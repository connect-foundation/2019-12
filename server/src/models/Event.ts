import {
  Table,
  Model,
  PrimaryKey,
  ForeignKey,
  Column,
  CreatedAt,
  UpdatedAt,
  DataType,
  BelongsTo,
  HasMany,
  AutoIncrement,
} from 'sequelize-typescript';
import { User } from './User';
import { TicketType } from './TicketType';

@Table({
  underscored: true,
})
export class Event extends Model<Event> {
  @AutoIncrement
  @PrimaryKey
  @Column(DataType.INTEGER)
  public id!: number;

  @ForeignKey(() => User)
  @Column(DataType.INTEGER)
  public userId!: number;

  @BelongsTo(() => User, 'userId')
  public user!: User;

  @Column(DataType.BOOLEAN)
  public isPublic!: boolean;

  @Column(DataType.STRING)
  public title!: string;

  @Column(DataType.DATE)
  public startAt!: Date;

  @Column(DataType.DATE)
  public endAt!: Date;

  @Column(DataType.STRING)
  public place!: string;

  @Column(DataType.STRING)
  public address!: string;

  @Column(DataType.STRING)
  public placeDesc!: string;

  @Column(DataType.STRING)
  public mainImg!: string;

  @Column(DataType.TEXT)
  public desc!: string;

  @CreatedAt
  public readonly createdAt!: Date;

  @UpdatedAt
  public readonly updatedAt!: Date;

  @HasMany(() => TicketType, 'eventId')
  public ticketTypes!: TicketType[];
}
