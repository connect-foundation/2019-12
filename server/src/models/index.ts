export { Event } from './Event';
export { Order } from './Order';
export { OrderTicket } from './OrderTicket';
export { TicketSubscription } from './TicketSubscription';
export { TicketType } from './TicketType';
export { User } from './User';
