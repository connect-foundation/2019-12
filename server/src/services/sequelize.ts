import { Sequelize } from 'sequelize-typescript';
import {
  Event,
  Order,
  OrderTicket,
  TicketSubscription,
  TicketType,
  User,
} from '../models';

const { DB_HOST, DB_USER, DB_PW, DB_NAME } = process.env;

export const sequelize = new Sequelize({
  host: DB_HOST,
  username: DB_USER,
  password: DB_PW || '',
  database: DB_NAME,
  dialect: 'mariadb',
  models: [Event, Order, OrderTicket, TicketSubscription, TicketType, User],
});
