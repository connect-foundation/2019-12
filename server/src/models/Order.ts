import {
  Table,
  Model,
  PrimaryKey,
  CreatedAt,
  UpdatedAt,
  Column,
  BelongsTo,
} from 'sequelize-typescript';
import { User } from './User';
import { Event } from './Event';

@Table({
  underscored: true,
})
export class Order extends Model<Order> {
  @PrimaryKey
  @Column
  public id!: number;

  @BelongsTo(() => User, 'user_id')
  public user!: User;

  @BelongsTo(() => Event, 'event_id')
  public event!: number;

  @CreatedAt
  public readonly createdAt!: Date;

  @UpdatedAt
  public readonly updatedAt!: Date;
}
