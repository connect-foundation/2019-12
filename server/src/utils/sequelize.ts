import { Sequelize } from 'sequelize-typescript';
import { Event, UserTicket, TicketType, User } from '../models';
import { readJSONData } from './readJSON';
import { resolve } from 'path';

const { DB_HOST, DB_PORT, DB_USER, DB_PW, DB_NAME } = process.env;
const SEED_DIR = resolve(__dirname, '../../db_seeds');

export const sequelize = new Sequelize({
  host: DB_HOST,
  port: DB_PORT ? +DB_PORT : 3306,
  username: DB_USER,
  password: DB_PW,
  database: DB_NAME,
  dialect: 'mariadb',
  models: [Event, UserTicket, TicketType, User],
});

export async function migrate() {
  console.info('DB migration start...');
  await sequelize.sync();
  console.info('DB migration end...');
}

export async function seed() {
  console.info('DB seed start...');
  // UserData 추가
  const usersPath = resolve(SEED_DIR, 'users.json');
  const usersData = await readJSONData<User>(usersPath);
  await User.bulkCreate(usersData);

  // EventData 추가
  const eventPath = resolve(SEED_DIR, 'events.json');
  const eventData = await readJSONData<Event>(eventPath);
  await Event.bulkCreate(eventData);

  // TicketData 추가
  const ticketPath = resolve(SEED_DIR, 'ticketTypes.json');
  const ticketData = await readJSONData<TicketType>(ticketPath);
  await TicketType.bulkCreate(ticketData);

  // OrderTikcetData 추가
  const userTicketPath = resolve(SEED_DIR, 'userTickets.json');
  const userTicketData = await readJSONData<UserTicket>(userTicketPath);
  await UserTicket.bulkCreate(userTicketData);

  console.info('DB seed end...');
}
