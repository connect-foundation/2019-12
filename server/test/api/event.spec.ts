import * as request from 'supertest';
import app from '../../src/app';
import { sequelize } from '../../src/utils/sequelize';
import { Event } from '../../src/models';

beforeAll(async () => {
  sequelize.options.logging = false;
  await sequelize.sync();
});

afterAll(() => {
  sequelize.close();
});

describe('Router / Events', () => {
  it('GET /api/events', async () => {
    await Promise.all([
      () => {
        const cnt = 2;
        request(app)
          .get('/api/events')
          .query({ cnt })
          .expect(200)
          .expect('Content-type', /application\/json/)
          .expect(res => expect(res.body).toHaveLength(cnt));
      },

      () => {
        const startAt = new Date('2018-04-30T10:00:00.000Z');
        request(app)
          .get('/api/events')
          .query({ startAt })
          .expect(200)
          .expect('Content-type', /application\/json/)
          .expect(res =>
            res.body.forEach((e: Event) =>
              expect(e.startAt.getTime()).toBeLessThan(startAt.getTime()),
            ),
          );
      },

      () => {
        const wrongId = 'wrong';
        request(app)
          .get('/api/events')
          .query({ lastId: wrongId })
          .expect(400)
          .expect('Content-type', /application\/json/);
      },
    ]);
  });
});
