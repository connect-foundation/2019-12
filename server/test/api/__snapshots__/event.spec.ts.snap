// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/events/:eventId/users 내 이벤트에 접근한 경우 200 1`] = `
Object {
  "email": "sdong001@gmail.com",
  "firstName": "성동",
  "id": 2,
  "lastName": "조",
  "phoneNumber": "01012341234",
}
`;

exports[`GET /api/events/:eventId/users 내 이벤트에 접근한 경우 200 2`] = `
Object {
  "id": 1,
  "isAttendance": true,
  "ticketTypeId": 2,
  "userId": 2,
}
`;
